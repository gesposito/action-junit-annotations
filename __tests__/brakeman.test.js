const { failingCasesFrom, annotationsFrom } = require("../index");

// Custom JUnit report built from Workflow jobs with
// https://github.com/gesposito/action-workflow-jobs-reporter

describe("Workflow report", () => {
  let failingCases;
  test("Converts JUnit XML to failing cases in JSON", async () => {
    failingCases = await failingCasesFrom("./test-results/brakeman.xml");
    expect(failingCases).toMatchSnapshot();
  });

  test("Converts failing cases in JSON to GitHub annotations", async () => {
    const annotations = annotationsFrom(failingCases);
    expect(annotations).toMatchSnapshot();
  });
});
