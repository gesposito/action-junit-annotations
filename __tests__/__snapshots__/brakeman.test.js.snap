// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workflow report Converts JUnit XML to failing cases in JSON 1`] = `
Array [
  Object {
    "classname": "Brakeman::CheckExecute",
    "failure": Object {
      "#text": "(High) Command Injection - Possible command injection near line 32 in app/controllers/users_controller.rb",
      "brakeman:code": "",
      "brakeman:confidence": "High",
      "brakeman:file": "app/controllers/users_controller.rb",
      "brakeman:fingerprint": "3949fc33d3d8408a7d4f3fb7d87e83ea771419e36833160218462b8c61d59135",
      "brakeman:line": "32",
      "message": "Possible command injection",
      "type": "Command Injection",
    },
    "name": "run_check",
    "path": "run_check",
    "time": "0",
  },
  Object {
    "classname": "Brakeman::CheckMassAssignment",
    "failure": Object {
      "#text": "(Medium) Mass Assignment - Specify exact keys allowed for mass assignment instead of using \`permit!\` which allows any keys near line 32 in app/controllers/posts_controller.rb: params.permit!",
      "brakeman:code": "",
      "brakeman:confidence": "Medium",
      "brakeman:file": "app/controllers/posts_controller.rb",
      "brakeman:fingerprint": "2f085da1d4b8bbc0e44f00c7a59ab8549640210dba219796aa63a6daead826d7",
      "brakeman:line": "32",
      "message": "Specify exact keys allowed for mass assignment instead of using \`permit!\` which allows any keys",
      "type": "Mass Assignment",
    },
    "name": "run_check",
    "path": "run_check",
    "time": "0",
  },
]
`;

exports[`Workflow report Converts failing cases in JSON to GitHub annotations 1`] = `
Array [
  Object {
    "annotation_level": "failure",
    "end_line": 0,
    "message": "(High) Command Injection - Possible command injection near line 32 in app/controllers/users_controller.rb",
    "path": "run_check",
    "start_line": 0,
  },
  Object {
    "annotation_level": "failure",
    "end_line": 0,
    "message": "(Medium) Mass Assignment - Specify exact keys allowed for mass assignment instead of using \`permit!\` which allows any keys near line 32 in app/controllers/posts_controller.rb: params.permit!",
    "path": "run_check",
    "start_line": 0,
  },
]
`;
